File: src/main/java/org/formation/model/Patient.java
package org.formation.model;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.formation.model.view.JsonViews;
import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonView;


@Entity
@DiscriminatorValue("patient")
public class Patient extends User {
	
	
	@DateTimeFormat(pattern="yyyy-MM-dd")
	@Temporal(TemporalType.DATE)
	private Date naissance;
	@JsonView(JsonViews.Common.class)
	private Integer secu;
	
	@OneToMany(mappedBy = "patient")
	@JsonView(JsonViews.Common.class)
	@JsonManagedReference
	private List<Rdv> listesRdv = new ArrayList<Rdv>();
	
	public Patient() {
	super();
	
	}

	public Date getNaissance() {
		return naissance;
	}

	public void setNaissance(Date naissance) {
		this.naissance = naissance;
	}

	public Integer getSecu() {
		return secu;
	}

	public void setSecu(Integer secu) {
		this.secu = secu;
	}

	public List<Rdv> getListesRdv() {
		return listesRdv;
	}

	public void setListesRdv(List<Rdv> listesRdv) {
		this.listesRdv = listesRdv;
	}

}

File: adoopteundoc/src/app/model/patient.ts
import {User} from './user';
import {Rdv} from './rdv';

export class Patient extends User{
  constructor(private _naissance?: Date, private _secu?: number, private _rdv?: Rdv[]) {
    super();
  }

  get naissance(): Date {
    return this._naissance;
  }

  set naissance(value: Date) {
    this._naissance = value;
  }

  get secu(): number {
    return this._secu;
  }

  set secu(value: number) {
    this._secu = value;
  }

  get rdv(): Rdv[] {
    return this._rdv;
  }

  set rdv(value: Rdv[]) {
    this._rdv = value;
  }
}
